### ISSUE TITLE
New Provider Api

### ISSUE THREAD

**Issue by gnoff** [gnoff/react-tunnel#4]
remove configuration props {forwardProvided, allowOverload}
Stop auto-providing arbitrary props on Provider

add new Provider prop: `provide: (parentProvided) => provided | plainObject`

if provide is a function it receives the the `provided` in context (only makes sense if this Provider is nested) and returns an object to be provided. It would be up to the caller to forward some/all/no parent provided values.

if the Provider is top level then parentProvided will be undefined

if provide is an object then that object will be provided in this provider. This means that to forward props from parent providers you must do so explicitly

Questions
1) should provided forwarding happen by default?

**Comment by gnoff** [comment]
### New API

#### Provider

``` js
<Provider provide={mapProvided | plainObject}>
```

##### provide = mapProvided([parentProvided])

``` js
mapProvided = (parentProvided) => {
   //... any logic, parentProvided could be undefined
  return {...anything};
}
```

##### provide = plainObject

Provider wraps `plainObject` in `defaultMapProvided`

``` js
defaultMapProvided = (parentProvided) => ({...parentProvided, ...plainObject})
```

ObjectMode auto forwards provided values and allows overloading

**Comment by gnoff** [comment]
Implemented in v0.1.0 aec29d854abfe2c9d1cc39197bf407d68a5c19b6
